### PR TITLE
Added --properties-allowed switch and retries to test.

### PR DESCRIPTION
**Pull request by ryanrolds**
This is part of reducing cardinality. The JUnit XML from Ginkgo and gotestsum contains arbitrary properties we don't want pushed to our metrics system.

* Adds new flag that allows defining an allowed list of properties
* Added retry package and used to improve test reliability
* Updates junit-go to latest version

### CODE REVIEW

**Comment by mdelapenya** [comment]
Let's rebase and resolve conflicts first 🙏

**Comment by ryanrolds** [comment]
I've rebased and resolved conflicts. This was the branch that originally included the retry changes. Are you wanting those removed?

**Comment by mdelapenya** [comment]
I think the branch still has conflicts, probably gotten from the retry, so yeah, let's remove them by now, as they sleep was not needed anymore with the wait.ForFile strategy in the tests

File: main_test.go [hunk 1]
```diff
@@ -251,25 +260,31 @@ func Test_Main_SampleXML(t *testing.T) {
...
 	// 2. remote white spaces
 	// 3. unmarshal each resource separately
 	// 4. assign each resource to the test report struct
-	content := string(jsonBytes)
 
-	jsons := strings.Split(strings.TrimSpace(content), "\n")
-	if len(jsons) != 2 {
-		t.Errorf("expected 2 JSONs, got %d - %s", len(jsons), jsons)
+	var jsons []string
+	err = retry.Do(func() error {
```
Comment (mdelapenya) [inline main_test.go:270]: Let's keep the wait strategy do its job here.

**Review by mdelapenya — CHANGES_REQUESTED** [review]

**Comment by ryanrolds** [comment]
The conflicts are from the PR that was marge after I resolved the conflicts from the earlier merge. 

The test is much faster without the 30s wait. I will resolve the conflicts in the morning.

**Comment by ryanrolds** [comment]
I tried it without the retry and I randomly get:
```
=== RUN   Test_Main_SampleXML
    main_test.go:187: could not get the collector port: port "4317/tcp" not found
2024/12/20 07:43:31 failed to upload metrics: failed to exit idle mode: dns resolver: missing port after port-separator colon
2024/12/20 07:43:31 traces export: failed to exit idle mode: dns resolver: missing port after port-separator colon
    main_test.go:236: expected 2 JSONs, got 1 - []
--- FAIL: Test_Main_SampleXML (5.62s)
panic: runtime error: index out of range [1] with length 1 [recovered]
	panic: runtime error: index out of range [1] with length 1
```

**Comment by ryanrolds** [comment]
I've resolved the merge conflicts and am switching the port retry to use `wait`.

File: main_test.go [hunk 2]
```diff
@@ -177,16 +176,15 @@ func setupRuntimeDependencies(t *testing.T) (context.Context, string, testcontai
 					ContainerFilePath: "/tmp/tests.json",
 				},
 			},
+			WaitingFor: wait.ForListeningPort("4317/tcp"),
```
Comment (mdelapenya) [inline main_test.go:179]: Neat!

File: README.md [hunk 3]
```diff
@@ -146,6 +146,7 @@ This tool is able to override the following attributes:
 | Service Name | --service-name | `junit2otlp` | Overrides OpenTelemetry's service name. If the `OTEL_SERVICE_NAME` environment variable is set, it will take precedence over any other value. |
 | Service Version | --service-version | Empty | Overrides OpenTelemetry's service version. If the `OTEL_SERVICE_VERSION` environment variable is set, it will take precedence over any other value. |
 | Trace Name | --trace-name | `junit2otlp` | Overrides OpenTelemetry's trace name. |
+| Properties Allowed | --properties-allowed | All | Comma separated list of properties to be allowed in the jUnit report. |
```
Comment (mdelapenya) [inline README.md:149]: question: as a non-native speaker, what is best: `properties-allowed` or `allowed-properties`?

Comment (ryanrolds) [reply README.md:149]: I prefer the subject before because if a "denied" is added they are grouped together when sorted alphabetically.

**Review by mdelapenya — APPROVED** [review]
LGTM, thanks!

**Pull request merged**